### PR TITLE
Badges

### PR DESCRIPTION
**Pull request by johnworth**
Adds schema definitions for Badges, Submissions, NewBadges, and NewSubmissions. Adds a dependency on org.flatland/ordered to get the repl to shut up after an upgrade to OpenJDK 11.

### CODE REVIEW

File: src/common_swagger_api/schema/badges.clj [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  (:import [java.util UUID]))
+
+(def SubmissionField (describe String "The JSON-encoded submission"))
+
+(s/defschema Submission
+  {:id
+   (describe UUID "The UUID for this submission")
+
+   :submission
+   SubmissionField})
```
Comment (slr71) [inline src/common_swagger_api/schema/badges.clj]: If you wanted to add some extra validation for the submission field, you can use the `AnalysisSubmission` schema defined in the `apps` service: https://github.com/cyverse-de/apps/blob/master/src/apps/routes/schemas/analysis.clj#L62-L115.

File: src/common_swagger_api/schema/badges.clj [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+(s/defschema NewSubmission
+  (dissoc Submission :id))
+
+(s/defschema Badge
+  {:id
+   (describe UUID "The UUID for the badge")
+
+   :user
+   (describe String "The username of the user that owns the object")
```
Comment (slr71) [inline src/common_swagger_api/schema/badges.clj]: This is completely optional, but you could define the user field to be of type `NonBlankString`, which is defined in `common-swagger-api.shcema`. Then if someone passed in an empty username, the schema validation would catch it for you. Of course, the username has to be matched against the users table anyway, so this extra level of validation isn't strictly necessary.

**Review by slr71 — APPROVED** [review]
I mentioned a couple of optional suggestions, but the code looks good to me. Feel free to :shipit: when ready.

File: src/common_swagger_api/schema/badges.clj (hunk 1 above)
Comment (johnworth) [reply src/common_swagger_api/schema/badges.clj]: This should be done now. Added the schema for AnalysisSubmission to common-swagger-api.schema.apps and used it here.

File: src/common_swagger_api/schema/badges.clj (hunk 2 above)
Comment (johnworth) [reply src/common_swagger_api/schema/badges.clj]: This should be done as well.

**Comment by johnworth** [comment]
Thanks for the review and suggestions.

**Pull request merged**